### ISSUE TITLE
Example in UserGuide on Repeating Regions raises error

### ISSUE THREAD

**Issue by hroemer** [mozilla/PyPOM#62]
I am following the example [Repeating Regions](https://github.com/mozilla/PyPOM/blob/master/docs/user_guide.rst#repeating-regions) in order to fill a repeated region in a page:

```python
from pypom import Page, Region
from selenium.webdriver.common.by import By

class Results(Page):
    _result_locator = (By.CLASS_NAME, 'result')

    @property
    def results(self):
        results = self.find_elements(*self._result_locator)
        return [self.Result(el) for el in results]

    class Result(Region):
        _name_locator = (By.CLASS_NAME, 'name')

        @property
        def name(self):
            return self.find_element(*self._name_locator).text
```

but end up in this error:

```python
def __init__(self, page, root=None):
>       super(Region, self).__init__(page.driver, page.timeout)
E       AttributeError: 'WebElement' object has no attribute 'driver'
```
This is because in the list comprehension that creates the individual region instances a `WebElement` is passed to the initializer.

Is this a documentation issue?

**Comment by hroemer** [comment]
Thank you for providing this awesome project and your quick reply, your suggestion works!
Here's the patch [user_guide_repeated_region.patch.zip](https://github.com/mozilla/PyPOM/files/1578995/user_guide_repeated_region.patch.zip). If you prefer a PR, please let me know.

One weird thing though, in my case I have a region that contains a list of repeated regions. Would you recommend:
```python
return [self.Result(self, el) for el in results]
```
over
```python
return [self.Result(self.page, el) for el in results]
```
in this case?
Both variants work, but it first example feels strange too me, as a `Region` is not a `Page` and only works since both objects contain the required `driver` and `timeout` attributes due to the `WebView` base class.

Maybe I am not familiar enough with the project, but to me it would make more sense if the `_init__` parameter for `Page` and `Region` would be named `view` instead of `page`.

**Comment by davehunt** [comment]
A pull request would be preferred, thanks!

The region expects a page in the constructor, so if you have a region that returns additional regions, you should pass `self.page` to them as per your second example. This will allow the eventual regions to access the page they belong to using `self.page`, which can be useful if you need to access sibling regions.

**Comment by hroemer** [comment]
I created a PR #63 including an example for nested regions. I hope it helps.

**Comment by hroemer** [comment]
Trying to understand the repeating regions example, I now also added a HTML snippet, as well. Maybe I totally misunderstand the intended goal, but it did not work out-of-the box either.
I had to change the `Result` instantiation code in order to make it work.

Please have a look at the user guide changes and the examples in my [forked branch](https://github.com/hroemer/PyPOM/tree/doc_issue_62/docs). These two pytest tests hopefully describe the expectations well enough:

```python
def test_pypom_repeated_regions(selenium: WebDriver):
    url = 'file:///{}'.format(Path(__file__).parent.joinpath("./repeated_regions.html").resolve())
    page = Results(selenium, base_url=url)
    page.open()
    assert len(page.results) == 4

def test_pypom_nested_regions(selenium: WebDriver):
    url = 'file:///{}'.format(Path(__file__).parent.joinpath("./nested_regions.html").resolve())
    page = MainPage(selenium, base_url=url)
    page.open()
    assert len(page.menu1.entries) == 5
    assert len(page.menu2.entries) == 3
```

**Comment by davehunt** [comment]
Fixed by babdbe36b7e2b78ca96019e5e89b8fa92d66a26e. Thanks @hroemer!
